### PR TITLE
Fix #3454: Merge admin pin activity

### LINKED ISSUE #3454: Merge admin_pin_activity into single xml file
Currently there are 4 versions of `admin_pin_activity.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by ishant904**
Fixes #3454 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

| Old SS                                                                                                                          | New SS                                                                                                                          |
|---------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------|
| Screenshot_1644776685 | Screenshot_1644950350 |
| Screenshot_1644776709 | Screenshot_1644950363 |
| Screenshot_1644776800 | Screenshot_1644950243 |
| Screenshot_1644776807 | Screenshot_1644950256 |

### CODE REVIEW

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 1]
```diff
@@ -112,7 +112,8 @@
         style="@style/StateButtonActive"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginTop="24dp"
+        android:layout_marginTop="@dimen/submit_button_margin_top"
+        android:layout_marginEnd="@dimen/submit_button_margin_end"
```
Comment (yash10019coder) [inline app/src/main/res/layout-land/admin_pin_activity.xml]: why you have added marginEnd here?

File: app/src/main/res/layout/admin_pin_activity.xml [hunk 2]
```diff
@@ -20,10 +20,10 @@
       android:layout_width="match_parent"
       android:layout_height="match_parent"
       android:background="@color/oppia_background_yellow_ivory"
-      android:paddingBottom="96dp">
+      android:paddingBottom="@dimen/admin_auth_container_padding_bottom">
 
       <View
-        android:layout_width="0dp"
+        android:layout_width="match_parent"
```
Comment (yash10019coder) [inline app/src/main/res/layout/admin_pin_activity.xml]: why we have match parent here?

**Review by rishidyno — COMMENTED** [review]
@ishant904 I can see that you have just added dimensions but haven't deleted any activity. Our main goal is to merge files into if possible into single XML so would suggest take another look at your PR and figure out a way to do that as what you have done is not correct.

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml [hunk 3]
```diff
@@ -36,14 +36,16 @@
...
         android:minWidth="360dp"
-        android:paddingBottom="148dp">
+        android:paddingBottom="@dimen/admin_auth_container_padding_bottom">
 
         <TextView
           android:id="@+id/admin_pin_main_text"
           style="@style/Subtitle1ViewStart"
           android:layout_width="0dp"
           android:layout_height="wrap_content"
+          android:layout_marginStart="@dimen/admin_pin_main_text_margin_start"
```
Comment (yash10019coder) [inline app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:46]: same goes here

**Comment by ishant904** [comment]
I was already working on that. In the phone layout the submit button is constrainted differently for land and port layouts.
For portrait -
`app:layout_constraintEnd_toEndOf="parent"`
For landscape -
`app:layout_constraintEnd_toEndOf="@id/admin_pin_input_confirm_pin"`
Can we change the constriant in this case. @rishidyno @yash10019coder @rt4914 let me know your thoughts.

**Review by yash10019coder — CHANGES_REQUESTED** [review]
@ishant904 left some comments PTAL thanks

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 1 above)
Comment (ishant904) [reply app/src/main/res/layout-land/admin_pin_activity.xml]: the button has a margin end of 28dp for portrait, so added margin end of 0dp for landscape. Let me know if you have a better approch for this.

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml (hunk 3 above)
Comment (ishant904) [reply app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:46]: same 0dp added here to match portrait layout xml

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 2 above)
Comment (ishant904) [reply app/src/main/res/layout/admin_pin_activity.xml]: the view doesnt share the width with any other view so 0dp and match parent should be the same. Also the same view has match parent in land layout. Please correct me if I am wrong.

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 4]
```diff
@@ -20,7 +20,7 @@
       android:layout_width="match_parent"
```
Comment (rishidyno) [inline app/src/main/res/layout-land/admin_pin_activity.xml]: @ishant904 please delete this file as we don't need it anymore.

**Review by rishidyno — CHANGES_REQUESTED** [review]

**Comment by rishidyno** [comment]
@ishant904 for tablet screens, I think we can remove the `LinearLayout` and can do the same implementation as what has been done for mobile screens and keep the original dimensions of tablet screens. Please let me know if this does not make sense or if you have any doubts. Also update the SS after changes. Thanks.

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 4 above)
Comment (ishant904) [reply app/src/main/res/layout-land/admin_pin_activity.xml]: @rishidyno can we ignore the submit button constraints. The constraints are different in port and land xml. Can you help me out in this

**Comment by ishant904** [comment]
@rishidyno yes we can remove linear layout and make it similar to phone layout but the width dimensions for textviews are a bit tricky. Some textviews have width 0dp(match_constraint) in tablet port and match parent in tablet land. Let me know if you can help me out in this.

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 2 above)
Comment (yash10019coder) [reply app/src/main/res/layout/admin_pin_activity.xml]: @ishant904  this view is using constraint layout and 0dp is match constaints so we should stick to 0dp as match constraint and match parent are different

**Review by yash10019coder — COMMENTED** [review]
Hi @ishant904 left a comment PTAL thanks

**Comment by ishant904** [comment]
@yash10019coder thanks I have reverted it. It would be great if you could help me in deleting the tablet layout. The layout width dimens for textviews are different in port and landscape mode. I have tried out different dimens and layouts but none of them seems to work as the overall UI changes.

**Comment by rt4914** [comment]
@yash10019coder Please check this PR

**Review by yash10019coder — COMMENTED** [review]
Hi @ishant904 sorry the delay, So for the tablet layouts you can create the dimens for the variable which exists in the layout file in dimens.xml of the specif configuration of tablet layout then you can use it in the layout.
There is no need to create the empty attributes for other files you can leave them think of it as a variable that you are creating and using at the required place only.
As the for the swp600 land there is mostly marginTop for which you can create the dimens and in swp600 port there's marginEnd so you can create the dimen for that too.
I hope this clarifies your doubt thanks.

**Comment by yash10019coder** [comment]
And also resolve the merge conflicts and merge with the latest develop thanks

**Pull request closed without merging**